src/main/java/core/basesyntax/MyTask.java

```diff
@@ -13,7 +15,35 @@ public MyTask(int startPoint, int finishPoint) {
 
     @Override
     protected Long compute() {
-        // write your code here
-        return null;
+        if (finishPoint - startPoint > 10) {
+            List<RecursiveTask<Long>> subTasks = new ArrayList<>(createSubTask());
+            for (RecursiveTask<Long> subTask : subTasks) {
+                subTask.fork();
+            }
+            Long result = 0L;
+            for (RecursiveTask<Long> subTask : subTasks) {
+                result += subTask.join();
+            }
+            return result;
+        } else {
+            if (startPoint >= finishPoint) {
+                return 0L;
+            }
+            if (startPoint == finishPoint - 1) {
+                return (long) startPoint;
+            }
+            MyTask myTask = new MyTask(startPoint + 1, finishPoint);
+            return myTask.compute() + startPoint;
```

Comment:
why do we need recursion here? read the description, it's much more simple

Reply:
recursion -> loop changed